### PR TITLE
Trying stuff for WC connection..v3

### PR DESCRIPTION
**Pull request by Velenir**
#267

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr273--dexreact.review.gnosisdev.com

File: src/api/wallet/WalletApiImpl.ts [hunk 1]
```diff
@@ -4,6 +4,103 @@ import assert from 'assert'
 import { getDefaultProvider } from '../'
 
 import WalletConnectProvider from '@walletconnect/web3-provider'
+import WalletConnectQRCodeModal from '@walletconnect/qrcode-modal'
+;(window as any).WalletConnectProvider = WalletConnectProvider
+
+class ETHProv extends WalletConnectProvider {
+  public send(payload: any, callback: any): any {
+    console.log('send::payload,callback', payload, callback)
+    return super.send(payload, callback)
```
[the remark this answers is not in the document]
Comment (Velenir) [reply src/api/wallet/WalletApiImpl.ts]: Yep, for debugging only

File: .babelrc [hunk 2]
```diff
@@ -1,5 +1,9 @@
 {
-  "presets": ["@babel/preset-env", "@babel/preset-react"],
+  "presets": [[
+    "@babel/preset-env",
+    {"exclude": ["@babel/plugin-transform-classes"]}
```
[the remark this answers is not in the document]
Comment (Velenir) [reply .babelrc]: There's a funny story behind that.

Both [TS with our target](https://www.typescriptlang.org/play/?target=1#code/MYGwhgzhAECC0FMAeAXBA7AJjAQtA3tAL4BQJ6CA7nABQCU0A9I3gJboSuYLSszBgQIBJmiRoAfQgBXAA4IATgDows2SACeNFAAs+AGjEKA5tIC2GFBDpA) and [babel with preset-env](https://babeljs.io/repl#?browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYGwhgzhAECC0FMAeAXBA7AJjAQtA3tAL4BQJ6CA7nABQCU0A9I3gJboSuYLSszBgQIBJmiRoAfQDmCFAAUATgHsUKgJ4AHBAHkAZjVh0AdGA0aQamigAWfADRiFUgK4BbDCgh0gA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-2%2Cenv&prettier=false&targets=&version=7.7.4&externalPlugins=) transpile classes to functions.
But `WalletConnectProvider` has a `new.target` in its constructor when it's compiled (can check the build file). So you get `constructor must be called with new` error.

Here I disable class transpilation, which is not a very good idea, so I'll change to monkey-patching prototype and will test again.
But for debugging it's much easier to have a class to override methods.

[the remark this answers is not in the document]
Comment (anxolin) [reply .babelrc]: aha, ok

**Comment by Velenir** [comment]
Closed in favor of #285

**Pull request closed without merging**